duckduckgo/Android
# Add 3 fire button animations and make it a setting

Task/Issue URL: https://app.asana.com/0/488551667048375/1198977345184094/f
Tech Design URL: 
CC: 

**Description**:
This PR brings to users a way to customize their experience on the Fire button. We are including 3 different animations (fire by default) and 1 option to disable it (based on users' feedback).

What changes are included?
- Add new 3 Fire button animations: Inferno, whirlpool, airstream
- Add an option to disable the animation
- Add a new option inside settings based on ✓ Design: Setting to select Fire button animation
- Add pixels approved during kickoff:
- Pixel which animation was used on each clear data
- Pixel when the user accesses new option
- Pixel when the user selects a new animation

**Steps to test this PR**:
**Test1: Default animation**
1. Fresh install
1. Clear data
1. Ensure Fire animation appears
1. Ensure pixel `m_fd_a` sent with params `fa=fai`

**Test2: Disable fire animation**
1. Go to settings
1. Click on Fire Button Animation
1. Ensure pixel `m_fas_o` sent
1. Select "None"
1. Ensure pixel `m_fas_s` sent with params `fa=fann`
1. Go back to the browser
1. Clear data
1. Ensure no Fire animation appeared
1. Ensure pixel `m_fd_a` sent with params `fa=fann`

**Test3: Whirlpool and Airstream animation**
1. Repeat Test 2 but choosing another animation
1. Ensure pixels are sent with expected params, same as in Test2 (whirlpool `fa=fawp`, airstream `fa=faas`)
1. Ensure expected animation executed when clearing data

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
